service/resource/deployment/resource.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/giantswarm/azuretpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/key"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "deployment"
+)
+
+type Config struct {
+	// Dependencies.
+
+	AzureConfig *client.AzureConfig
+	Logger      micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new resource by
+// best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		AzureConfig: nil,
+		Logger:      nil,
+	}
+}
+
+type Resource struct {
+	// Dependencies.
+
+	azureConfig *client.AzureConfig
+	logger      micrologger.Logger
+}
+
+// New creates a new configured deploy resource.
+func New(config Config) (*Resource, error) {
+	// Dependencies.
+	if config.AzureConfig == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.AzureConfig must not be empty.")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty.")
+	}
+
+	newService := &Resource{
+		azureConfig: config.AzureConfig,
+		logger: config.Logger.With(
+			"resource", Name,
+		),
+	}
+
+	return newService, nil
+}
+
+// GetCurrentState gets the current deployments for this cluster from the
+// Azure API.
+func (r *Resource) GetCurrentState(obj interface{}) (interface{}, error) {
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	resourceGroupName := key.ClusterID(customObject)
+	deploymentClient, err := r.getDeploymentsClient()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var deployments []Deployment
+	{
+		for _, deploymentName := range getDeploymentNames() {
+			deploymentExtended, err := deploymentClient.Get(resourceGroupName, deploymentName)
+			if err != nil {
+				if client.ResponseWasNotFound(deploymentExtended.Response) {
+					// Fall through.
+					continue
+				}
+
+				return nil, microerror.Mask(err)
+			}
+
+			deployment := Deployment{
+				Name:          *deploymentExtended.Name,
+				Parameters:    *deploymentExtended.Properties.Parameters,
+				ResourceGroup: resourceGroupName,
+				Template:      *deploymentExtended.Properties.Template,
+			}
+			deployments = append(deployments, deployment)
+		}
+	}
+
+	return deployments, nil
+}
+
+// GetDesiredState is not yet implemented.
+func (r *Resource) GetDesiredState(obj interface{}) (interface{}, error) {
+	return []Deployment{}, nil
+}
+
+// GetCreateState is not yet implemented.
+func (r *Resource) GetCreateState(obj, currentState, desiredState interface{}) (interface{}, error) {
+	return []Deployment{}, nil
+}
+
+// GetDeleteState returns an empty deployments collection. Deployments and the
+// resources they manage are deleted when the Resource Group is deleted.
```

Comment:
As per this comment deleting deployments will not be implemented. Instead, we'll delete the resource group to ensure all resources are deleted.